src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CAConfigurationMonitor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.certificate;
+
+import com.aws.greengrass.clientdevices.auth.api.DomainEvents;
+import com.aws.greengrass.clientdevices.auth.api.Result;
+import com.aws.greengrass.clientdevices.auth.certificate.events.CAConfigurationChanged;
+import com.aws.greengrass.clientdevices.auth.connectivity.ConnectivityInformation;
+import com.aws.greengrass.clientdevices.auth.exception.CertificateGenerationException;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import javax.inject.Inject;
+
+
+// TODO: This class is acting almost like a handler the reason for not using a handler directly is because it registers
+//  CertificateGenerators same as the CertExpiryMonitor and CISMonitor. In order to minimize the surface area of changes
+//  without refactoring how we store subscribers we are taking this approach. Once we refactor how we store handlers
+//  this should become a fully fledged handler.
+public class CAConfigurationMonitor {
```

Comment:
In order to avoid doing more refactoring (for now), I am taking this hybrid approach of following what we currently do to register external subscribers that get their certificates rotated (using monitor classes) while at the same time using domain events.

I refrained from creating a listener and a handler due to the fact that we need a way to store the generators and I didn't want how we store the generators for all the monitors yet (it is explained in the TODO)